scripts/parse-arguments.sh

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bash
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+set -ex
+TEST_ROOT_DIR=$(git rev-parse --show-toplevel)
+BRANCH="master"
+SPARK_REPO="https://github.com/apache/spark"
+SPARK_REPO_LOCAL_DIR="$TEST_ROOT_DIR/target/spark"
+UNPACKED_SPARK_TGZ="$TEST_ROOT_DIR/target/spark-dist-unpacked"
+IMAGE_TAG_OUTPUT_FILE="$TEST_ROOT_DIR/target/image-tag.txt"
+DEPLOY_MODE=minikube
+IMAGE_REPO="docker.io/kubespark"
+SKIP_BUILDING_IMAGES=false
+SPARK_TGZ="N/A"
```

Comment:
This is less than ideal. It's more elegant to have here: `SPARK_TGZ=` and then to check in other places, `if [[ -z $SPARK_TGZ ]]. The problem is that Maven's `exec` plugin has to pass in a fixed set of command line arguments. Maven will always have to pass in some `spark-tgz` argument. But we can't pass in empty, because we then end up with commands like `setup-integration-test-env --spark-tgz --deploy-mode cloud` which screws up the argument shifting.

An alternative we can employ in general is to allow these variables to be set by environment variables as well and to pass them in as environment variables from Maven. Ideas are welcome here.

Reply:
You could use the maven exec plugin like this:

```
              <executable>/bin/sh</executable>
              <arguments>
                <argument>-c</argument>
                <argument>arbitrary length here ; another command</argument>
              </arguments>
```

Reply:
The problem is that the arguments themselves have to be fixed. I can't sometimes pass in the `--spark-tgz` flag and sometimes not pass the `--spark-tgz` flag.